src/bgen/Generator.cs

```diff
@@ -3324,7 +3324,7 @@ void GenerateTypeLowering (MethodInfo mi, bool null_allowed_override, out String
 						by_ref_init.Insert (0, string.Format ("NSArray? {0}ArrayValue = {0} is null ? null : NSArray.FromStrings ({0});\n", pi.Name.GetSafeParamName ()));
 						by_ref_init.AppendFormat ("{0}ArrayValue is null ? NativeHandle.Zero : {0}ArrayValue.Handle;\n", pi.Name.GetSafeParamName ());
 					} else if (isNSObject || isINativeObjectSubclass) {
-						by_ref_init.AppendFormat ("{0} is null ? NativeHandle.Zero : {0}.Handle;\n", pi.Name.GetSafeParamName ());
+						by_ref_init.AppendFormat ("Runtime.RetainAndAutoreleaseNSObject ({0});\n", pi.Name.GetSafeParamName ());
```

Comment:
Looks like you need to use `RetainAndAutoreleaseNativeObject` instead:

```suggestion
						by_ref_init.AppendFormat ("Runtime.RetainAndAutoreleaseNativeObject ({0});\n", pi.Name.GetSafeParamName ());
```